oneview/data_source_logical_interconnect.go

```diff
@@ -0,0 +1,435 @@
+							Optional: true,
+						},
+						"id": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"igmp_idle_timeout_interval": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"interconnect_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"mac_refresh_interval": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"modified": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"state": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"status": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"type": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"uri": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"fusion_domain_uri": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"interconnect_map": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"interconnect_map_entries": {
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"enclosure_index": {
+										Type:     schema.TypeInt,
+										Optional: true,
+									},
+									"interconnect_uri": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"location": {
+										Type:     schema.TypeList,
+										Optional: true,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"location_entries": {
+													Type:     schema.TypeList,
+													Optional: true,
+													Elem: &schema.Resource{
+														Schema: map[string]*schema.Schema{
+															"type": {
+																Type:     schema.TypeString,
+																Optional: true,
+															},
+															"value": {
+																Type:     schema.TypeString,
+																Optional: true,
+															},
+														},
+													},
+												},
+											},
+										},
+									},
+									"logical_downlink_uri": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+									"permitted_interconnect_type_uri": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			"interconnects": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+			"logical_interconnect_group_uri": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"modified": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"name": {
```

Comment:
I think name is a required parameter as we are using name as the filter to get the particular resource.